grease-socket/src/lib.rs

```diff
@@ -27,8 +32,8 @@ extern crate rand;
 use std::collections::{HashMap, VecDeque};
 use std::convert::From;
 use std::fmt;
-use std::io::prelude::*;
 use std::io;
+use std::io::prelude::*;
```

Comment:
When I build in 1.26, it tells me this io::prelude import is complete redundant.

Reply:
I'm surprised by that - builds ok for me, and if I remove it, this happens:
```diff
diff --git a/grease-socket/src/lib.rs b/grease-socket/src/lib.rs
index 673cde7..21dbb05 100644
--- a/grease-socket/src/lib.rs
+++ b/grease-socket/src/lib.rs
@@ -33,7 +33,6 @@ use std::collections::{HashMap, VecDeque};
 use std::convert::From;
 use std::fmt;
 use std::io;
-use std::io::prelude::*;
 use std::net;
 use std::thread;
```
```
PS C:\Users\david\Devel\grease\grease-socket> rustc --version
rustc 1.27.0-nightly (565235ee7 2018-05-07)
PS C:\Users\david\Devel\grease\grease-socket> cargo build
   Compiling grease-socket v0.2.0 (file:///C:/Users/david/Devel/grease/grease-socket)
error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:476:24
    |
476 |             match cs.connection.write(&pw.data[pw.sent..]) {
    |                                 ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error[E0599]: no method named `read` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:526:25
    |
526 |                 match cs.connection.read(buffer.as_mut_slice()) {
    |                                     ^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Read;
    |

error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:666:25
    |
666 |                 match cs.connection.write(&req_send.data) {
    |                                     ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `grease-socket`.

To learn more, run the command again with --verbose.
```

and under WSL:
```
root@david-laptop:/mnt/c/Users/david/Devel/grease/grease-socket# rustc --version
rustc 1.26.0 (a77568041 2018-05-07)
root@david-laptop:/mnt/c/Users/david/Devel/grease/grease-socket# cargo build
   Compiling grease-socket v0.2.0 (file:///mnt/c/Users/david/Devel/grease/grease-socket)
error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:476:24
    |
476 |             match cs.connection.write(&pw.data[pw.sent..]) {
    |                                 ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error[E0599]: no method named `read` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:526:25
    |
526 |                 match cs.connection.read(buffer.as_mut_slice()) {
    |                                     ^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Read;
    |

error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:666:25
    |
666 |                 match cs.connection.write(&req_send.data) {
    |                                     ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `grease-socket`.

To learn more, run the command again with --verbose.
```

Reply:
Ah, you mean line 788 in the tests

Reply:
Fixed now